doc/MANUAL.adoc

```diff
@@ -929,6 +930,33 @@ Optional attributes:
 * *update-mtime*: If *true*, update the modification time (mtime) of cache
   entries that are read. The default is *false*.
 
+=== Redis storage backend
+
+URL format: `redis://HOST:PORT` or `redis://UNIX_SOCK`
+
+This backend stores data in a Redis (or Redis-compatible) server.
+There are implementations for both memory and disk based storage.
+
+See <https://redis.io>
+
+Note that ccache will not perform any cleanup of the Redis storage.
+But you can configure LRU eviction: <https://redis.io/topics/lru-cache>
+
+In addition to the in-memory database, there are also persistent NoSQL.
+See <https://github.com/yinqiwen/ardb> (using https://rocksdb.org/[RocksDB] or similar)
+
+Examples:
+
+* `redis://localhost:6379`
+* `redis:///var/run/redis/redis-server.sock`
+
+Optional attributes:
+
+* *connect-timeout*: Timeout (in ms) for network connection.
+* *operation-timeout*: Timeout (in ms) for Redis commands.
```

Comment:
What do you think about my [suggested default values](https://github.com/ccache/ccache/issues/859#issuecomment-869208683)?

I think that we should have (quite small) default values since it makes sense to me to time out quickly for a misbehaving network or remote server. I also think we don't need to support "no timeout".

Reply:
I'm not sure about the values to use, but the code should be ready for it (tested it locally, even if there are no tests)

Reply:
Added 100 ms connection timeout, and 10 second operation timeout (in case of an overloaded shared server)